Cognite.Simulator.Utils/FileStorageClient.cs

```diff
@@ -29,38 +33,65 @@ public FileStorageClient(HttpClient client, ILogger<FileStorageClient> logger)
         /// Downloads the file from the provided <paramref name="uri"/> and saves
         /// it in the provided <paramref name="filePath"/>
         /// </summary>
-        /// <param name="uri"></param>
-        /// <param name="filePath"></param>
+        /// <param name="uri">URI to download from</param>
+        /// <param name="filePath">Path to save the file</param>
+        /// <param name="failLargeFiles">Set to true to throw an exception if the file is larger than <see cref="LargeFileSize"/></param>
         /// <returns><c>true</c> if success, else <c>false</c></returns>
-        public async Task<bool> DownloadFileAsync(Uri uri, string filePath)
+        public async Task<bool> DownloadFileAsync(Uri uri, string filePath, bool failLargeFiles = false)
         {
             try
             {
-                var response = await _client.GetAsync(uri)
+                var response = await _client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead)
                     .ConfigureAwait(false);
                 if (!response.IsSuccessStatusCode)
                 {
                     _logger.LogWarning("Failed to download the file: {Message}", response.ReasonPhrase);
                     return false;
                 }
 
+                if (response.Content.Headers.ContentLength > MaxFileDownloadSize)
+                {
+                    _logger.LogWarning("File size exceeds the maximum allowed size: {Size} bytes", MaxFileDownloadSize);
+                    return false;
+                }
+
+                if (failLargeFiles && response.Content.Headers.ContentLength > LargeFileSize)
+                {
+                    throw new ConnectorException(
+                        $"The file is larger than {LargeFileSize / 1024 / 1024} MB and needs to be pre-downloaded before any operations can be performed." +
+                        "Connector hasn't been able to pre-download the file yet. " +
+                        "Please wait for the file to be downloaded and try again."
```

Comment:
The error says `Please wait for the file to be downloaded and try again.` But actually if `failLargeFiles` is true, the connector would keep throwing the same exception on retrying? so should we remove the `try again` part

Reply:
no, this only fails when the file isn't downloaded beforehand and there is a simulation run that tries to use it.
in that case we normally try to download i on a fly, remember?
however, this doesn't make sense to keep when files are big, because they take time to be downloaded, so the simulations will get stuck for a long time. so better to fail it quick and let the library download it later

Reply:
ref this `.DownloadFileAsync(uri, filename, failLargeFiles: isTemporary)`